### ISSUE TITLE
Feature Request: Import GIF pictures

### ISSUE THREAD

**Issue by 7flash** [excalidraw/excalidraw#5301]

**Comment by anuzk13** [comment]
Would find this helpful too~

**Comment by eric-schleicher** [comment]
Specifically Animated Gifs would be helpful.

**Comment by ctwhome** [comment]
when creating maps of contents for learning sessions, it is fundamental to be able play the gifs

**Comment by EllyKher** [comment]
@dwelle @ad1992 @lipis 
Any chance this could be turned into a feature? :) Have you guys considered it?

**Comment by appei** [comment]
If anyone needs a workaround for this right now. Copy and paste [this HTML](https://codepen.io/appuufish/pen/rNovgmq) into a notepad doc, replace the url with your gif url , save the notepad as .html (remember to click save as all file types when you do this and write '.html' at the end of your file name), drag the html file into ur browser, copy the url in the browser and paste it into excalidraw as a web embed.  BOOM resizable gif.

**Comment by ctwhome** [comment]
So it should be relatively easy to just add all images gif as embed HTML. Thanks for the tip.

**Comment by eric-schleicher** [comment]
@appei i'm trying to reproduce your steps.

save the file locally with .html.   
open it in browser (file:///C:/Users/user/Desktop/animated-gif-excalidraw.html)
 and then add a web embed and set that to the link-url?

![image](https://github.com/excalidraw/excalidraw/assets/8261783/bbe53658-821c-4d95-99dd-d8772358b164)

I very much like the scaling part.

**Comment by appei** [comment]
> very much like the scaling part, but not quite sure how that would work (because web embed whitelisting) and also because of local file resou

Yeah, unfortunately, this would only work locally unless you create and host a website using the html code I sent. Then use that URL to embed. For me, I use Excalidraw as a journal so hosting and sharing isn't as big a deal for me. But if it is for you I think you might have to do it how it's listed in the last update and just deal with black bars. Or go to the trouble of hosting websites with a single gif.

**Comment by k88936** [comment]
2 years.... if we were adding this feature, what would be the best place ? 
to reuse the insert image and route gif here or  
have a complete new tool naming "animation" like this ?
<img width="567" height="216" alt="Image" src="https://github.com/user-attachments/assets/906144b9-5c2e-4c46-aa1d-e1b2d2c34011" />

**Comment by Woolfer0097** [comment]
@k88936 definitely first option (reuse the insert image and route gif here)
do you developing this issue now?

**Comment by k88936** [comment]
> [@k88936](https://github.com/k88936) definitely first option (reuse the insert image and route gif here) do you developing this issue now?
yeah, I am working on this for my personal needs now.

my current impl is also to insert a web embed,  draw the gif via browser beyond the canva. ( just like the approach above in the issue )  

At first , I am afraid that

1.  the first approach will bring much complexity to the code base,
2. more source or format support  ( but now I think gif is simple enough, other format can be converted externally...)

so I created a new tool.

----

 I will make it simply the first option, and  raise a WIP pr later.

**Comment by k88936** [comment]
or we need also support webp anim and apng format ?

**Comment by k88936** [comment]
@Woolfer0097  If we are going to support animated webp , we can not simply judge if it is a static image or a piece of anim by its extention.      the first way will be complicated
